### ISSUE TITLE
Update new python source tutorial to reflect new development instructions

### ISSUE THREAD

**Issue by sherifnada** [airbytehq/airbyte#2471]
## Tell us about the documentation you'd like us to add or update
remove references to gradle commands in line with #2360

**Comment by Wang-Yong2018** [comment]
I would like to airbyte to update minimal python3 version in the tech stack spec or  https://docs.airbyte.io/tutorials/building-a-python-source
- https://docs.airbyte.io/architecture/tech-stack
- https://docs.airbyte.io/tutorials/building-a-python-source

At present, if the python version is  3.6 or lower, the the python-source tutorial will be failed at step3 run source using python.
The code expection will be throwed:

> from __future__ import annotations model could not be found.

After google search, It seems above function was introduced after python 3.7. So later, I upgrade to Python 3.9. The tutorial could be go on.

I double check the documations and found there is no place clarified python3 minimum version.  Though in the git root folder, there is .python-version file existed.

To sum up, clarify minimum python version in tech stack spec and python -source tutorial documents.  It can help airbyte client reduce time of learning curve and speed up the process of trial of airbyte.

Thanks

WangYong

**Comment by sherifnada** [comment]
@Wang-Yong2018 thanks for the great suggestion. I've created a PR to update the docs here: https://github.com/airbytehq/airbyte/pull/2743

**Comment by Wang-Yong2018** [comment]
thanks
